### PR TITLE
RFC-3: Add protocol basic

### PR DESCRIPTION
**Pull request by npofsi**
issue: #1

### CODE REVIEW

File: credential.go [hunk 1]
```diff
@@ -30,6 +30,10 @@ const (
 	// Storage service like gcs will take token files as input, we provide base64 protocol so that user
 	// can pass token binary data directly.
 	ProtocolBase64 = "base64"
+	// ProtocolBasic will hold user and password credential.
+	//
+	// value = [User, Password]
```
Comment (Xuanwo) [inline credential.go]:
```suggestion
	// value = [user, password]
```

File: docs/rfcs/1-add-protocol-basic.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
+- Author: npofsi [npofsi@outlook.com](mailto:npofsi@outlook.com)
+- Start Date: 2021-08-02
+- RFC PR: N/A
```
Comment (Xuanwo) [inline docs/rfcs/1-add-protocol-basic.md]:
```suggestion
- RFC PR: https://github.com/beyondstorage/go-credential/pull/3
```

File: docs/rfcs/1-add-protocol-basic.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
+- Author: npofsi [npofsi@outlook.com](mailto:npofsi@outlook.com)
```
Comment (Xuanwo) [inline docs/rfcs/1-add-protocol-basic.md]:
```suggestion
- Author: npofsi <npofsi@outlook.com>
```

File: docs/rfcs/1-add-protocol-basic.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+- Start Date: 2021-08-02
+- RFC PR: N/A
+- Tracking Issue: [beyondstorage/go-credential#1](https://github.com/beyondstorage/go-credential/issues/1)
+
+# RFC-1:
```
Comment (Xuanwo) [inline docs/rfcs/1-add-protocol-basic.md]:
```suggestion
# RFC-3: Add protocol basic
 ```

**Comment by Xuanwo** [comment]
The normal RFC process requires a separate RFC PR and implement PR. It's OK to merge them together as it's a simple RFC. But next time, please send RFC PRs first.

**Comment by npofsi** [comment]
> 
> 
> The normal RFC process requires a separate RFC PR and implement PR. It's OK to merge them together as it's a simple RFC. But next time, please send RFC PRs first.

Ok, I'll remember it.

**Review by Xuanwo — APPROVED** [review]

File: credential.go [hunk 5]
```diff
@@ -119,6 +135,8 @@ func Parse(cfg string) (Credential, error) {
 		return NewEnv(), nil
 	case ProtocolBase64:
 		return NewBase64(s[1]), nil
+	case ProtocolBasic:
```
Comment (Xuanwo) [inline credential.go:138]: Please add related unit test cases for basic protocol.

**Pull request merged**